src/rockstor/storageadmin/static/storageadmin/js/views/samba.js

```diff
@@ -104,10 +104,16 @@ SambaView = RockstorLayoutView.extend({
     },
 
     switchStatus: function(event, state) {
-        if (state) {
-            this.startService();
+        if (!this.service.get('config')) {
```

Comment:
Unfortunately, the samba service is an exception to the other services as its `'config'` attribute is not `null` but an empty json string when not configured. This is due to:
https://github.com/rockstor/rockstor-core/blob/ba0ee33b7d2cace4e2438362be1cd84b9c84096e/src/rockstor/system/samba.py#L182-L216

We would thus need to either change the above, or simply change how we check for an unconfigured smb service here:
```suggestion
        if (_.isEmpty(JSON.parse(this.service.get('config')))) {
```

Reply:
It may be worth noting that the services page may also be an issue here then.

https://github.com/rockstor/rockstor-core/blob/fd3d07236cd26d968b00ec55d97c10ca4786c9b0/src/rockstor/storageadmin/static/storageadmin/js/views/services.js#L137-L147

The if statement there has the same exact check. If I'm understanding that correctly, it wouldn't work either way then, and only ever opens when configuring samba manually by clicking the wrench icon.

If that's the case, it may be best to change the return value to match the other services. This would fix it and keep it consistent with the others.

Since it's python, start as `config = None`, and then swap it to a dictionary right before adding something to it:

```python
if config is None:
    config = {}
config[fields[0].strip()] = fields[1].strip() 
```

I'm not sure if there would be any other ramifications to that though. Does this check happen anywhere else for samba?

Reply:
Very good points and I completely agree with you.

As to why that happens, I believe it "simply" is a matter of order. Note, however, that all I'm going to write is from my current understanding so still needs to be verified before I can say for sure. Indeed, in a brand new Rockstor install, the `config` field is empty (null), and remains as such when the user visits the services page. But, upon visiting the Storage > Samba page, `get_global_config()` is triggered and save `{}` as config in the model.

I believe your approach is far superior to what I originally proposed and would allow us to stay consistent across pages and services.

> I'm not sure if there would be any other ramifications to that though. Does this check happen anywhere else for samba?

That's exactly why I originally was afraid of touching `get_global_config()` in the first place but because this seems to be the only place it's called, it should be ok to apply your suggestions to it. We'll need to be more careful with testing it, though of course.

Thanks for catching this.